pkg/evaluators/authorization/json_test.go

```diff
@@ -251,33 +229,33 @@ func TestCall(t *testing.T) {
 
 	// multiple rules with at least one unauthorized
 	jsonAuth = &JSONPatternMatching{
-		Rules: []json.JSONPatternMatchingRule{
-			{
+		Rules: jsonexp.All(
+			jsonexp.Pattern{
 				Selector: "context.request.http.headers.x-secret-header",
-				Operator: "eq",
+				Operator: jsonexp.EqualOperator,
 				Value:    "no-one-knows",
 			},
-			{
+			jsonexp.Pattern{
 				Selector: "context.request.http.headers.x-secret-header",
-				Operator: "neq",
+				Operator: jsonexp.NotEqualOperator,
 				Value:    "no-one-knows",
 			},
-			{
+			jsonexp.Pattern{
 				Selector: "auth.metadata.letters",
-				Operator: "incl",
+				Operator: jsonexp.IncludesOperator,
 				Value:    "xxxxx",
 			},
-			{
+			jsonexp.Pattern{
 				Selector: "auth.metadata.letters",
-				Operator: "incl",
+				Operator: jsonexp.IncludesOperator,
 				Value:    "c",
 			},
-			{
+			jsonexp.Pattern{
```

Comment:
These tests look good, but wondering is there an exsting or needed test around nested any and all patterns?

Reply:
I might be missing someting, but I don't see test cases around `any` ??

Reply:
I see there are more test cases below in a different file